cloud-controller-manager/osc/osc_fakes.go

```diff
@@ -108,26 +110,26 @@ type FakeCompute interface {
 
 // FakeComputeImpl is an implementation of the FakeEC2 interface used for testing
 type FakeComputeImpl struct {
-	aws                      *FakeOscServices
-	Subnets                  []*ec2.Subnet
-	DescribeSubnetsInput     *ec2.DescribeSubnetsInput
-	RouteTables              []*ec2.RouteTable
-	DescribeRouteTablesInput *ec2.DescribeRouteTablesInput
+	osc                      *FakeOscServices
+	Subnets                  []osc.Subnet
+	DescribeSubnetsInput     *osc.ReadSubnetsRequest
+	RouteTables              []osc.RouteTable
+	DescribeRouteTablesInput *osc.ReadRouteTablesRequest
 }
 
 // ReadVms returns fake instance descriptions
-func (ec2i *FakeComputeImpl) ReadVms(request *ec2.DescribeInstancesInput) ([]*ec2.Instance, error) {
-	matches := []*ec2.Instance{}
-	for _, instance := range ec2i.aws.instances {
-		if request.InstanceIds != nil {
-			if instance.InstanceId == nil {
+func (ec2i *FakeComputeImpl) ReadVms(request *osc.ReadVmsRequest) ([]osc.Vm, error) {
+	matches := []osc.Vm{}
+	for _, instance := range ec2i.osc.instances {
+		if request.GetFilters().VmIds != nil {
+			if instance.VmId == nil {
 				klog.Warning("Instance with no instance id: ", instance)
 				continue
 			}
 
 			found := false
-			for _, instanceID := range request.InstanceIds {
-				if *instanceID == *instance.InstanceId {
+			for _, instanceID := range request.Filters.GetVmIds() {
```

Comment:
why there is both request.Filters.GetVmIds()  and request.GetFilters().VmIds ?

Reply:
The first one (l124) is to check the pointer, the second one is to retrieve the value with native functions